### ISSUE TITLE
[Bug]: Compatibility with GA4 consent and services

### ISSUE THREAD

**Issue by MelCCI** [orestbida/cookieconsent#652]
### Expected Behavior

You can choose between different categories, and only the categories you select will appear as granted in the Google Analytics consent mode.
![image](https://github.com/orestbida/cookieconsent/assets/57983830/b4ae97bd-83a6-42f6-a810-d520db691baf)
![image](https://github.com/orestbida/cookieconsent/assets/57983830/0ba24398-9e33-425f-ab05-9e10d06e962a)
![image](https://github.com/orestbida/cookieconsent/assets/57983830/11a8d416-617b-41f4-aa1a-0fb485a34e4e)

### Current Behavior

Even if you check only certain categories, they will all appear in the modes consent because this does not filter out the cookie.services section.
![image](https://github.com/orestbida/cookieconsent/assets/57983830/4f255bf3-d5db-4afe-9ece-720c3c092df4)
![image](https://github.com/orestbida/cookieconsent/assets/57983830/a5751931-3183-47ea-825e-a481c977f658)
![image](https://github.com/orestbida/cookieconsent/assets/57983830/c981a515-1d9a-499f-9e15-b00e03adeb12)

### Steps to reproduce

Setup following CookieConsent, then configure Google Analytics with consent mod depending on cc_cookie

CookieConsent.run({
    guiOptions: {
        consentModal: {
            layout: "box",
            position: "bottom left",
            equalWeightButtons: true,
            flipButtons: true
        },
        preferencesModal: {
            layout: "box",
            position: "right",
            equalWeightButtons: true,
            flipButtons: false
        }
    },
    categories: {
        necessary: {
            readOnly: true
        },
        analytics: {},
        ads: {}
    },
    language: {
        default: "fr",
        autoDetect: "browser",
        translations: {
            fr: {
                consentModal: {
                    description: "Chez Stores-et-rideaux.com, nous utilisons les cookies nécessaires pour faciliter votre navigation.",
                    acceptAllBtn: "ACCEPTER ET FERMER",
                    acceptNecessaryBtn: "Refuser",
                    showPreferencesBtn: "Personnaliser"
                },
                preferencesModal: {
                    title: "Choisir les cookies à accepter",
                    acceptAllBtn: "Tout Accepter",
                    acceptNecessaryBtn: "Refuser",
                    savePreferencesBtn: "Accepter et fermer",
                    closeIconLabel: "Close modal",
                    serviceCounterLabel: "Service|Services",
                    sections: [
                        {
                            title: "Essentiels",
                            linkedCategory: "necessary"
                        },
                        {
                            title: "Analytics",
                            linkedCategory: "analytics"
                        },
                        {
                            title: "Ads",
                            linkedCategory: "ads"
                        },
                    ]
                }
            }
        }
    });
});

### Proposed fix or additional info.

I finished by create a new cookie, to edit it myself and remove the services part. I think that google see the services part and use it too for gtag consent mod. When I remove manually the service part in cc_cookie it works fine (thats's why I'm able to reproduce it)

I've seen stories everywhere about services for categories, but nothing about the general service that encompasses all categories. I based myself on the github and the doc.
https://cookieconsent.orestbida.com/reference/api-reference.html

### Version

3.0.0-rc.15

### On which browser do you see the issue?

Chrome

**Comment by orestbida** [comment]
Not sure I understand. How did you set up gtag? What does the plugin's cookie have to do with gtag?

**Comment by orestbida** [comment]
> Do you mean that we manage the sending of gtag with granted ourselves? 

Yes, since the plugin doesn't come with any pre-configured service. The fact that you are actually getting some consent info means that you are actually loading the gtag script.

Also, there is no relation between the plugin's own cookie (cc_cookie by default) and the ones gtag/ga4 reads, so I don't see how you're getting a list of the accepted categories in the first place.

[Here](https://github.com/orestbida/cookieconsent/discussions/634#discussioncomment-8373448) is an example setup of GTM with google consent mode v2.

You might have to change a few things, based on your needs. For example, if you don't use google ads, simply delete the script with the `data-service="Google Ads"` attribute.

**Comment by MelCCI** [comment]
Thank you for the quick answers to my problem. I managed to find the solution.

For those of you in the same situation, I took the problem the other way round and looked at the GA4 settings.

I changed the data search to a regular expression that only takes into account items inserted in categories

![image](https://github.com/orestbida/cookieconsent/assets/57983830/8e5a2be4-df76-4003-94e8-3f31b5d326b8)

Thanks for your help
